dali/python/nvidia/dali/pickling.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2020-2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import inspect
+import pickle
+import sys
+import types
+import marshal
+import importlib
+import io
+
+dummy_lambda = lambda : 0
+
+def set_funcion_state(fun, state):
+    fun.__globals__.update(state['global_refs'])
+    fun.__defaults__ = state['defaults']
+    fun.__kwdefaults__ = state['kwdefaults']
+
+def function_unpickle(name, qualname, code, closure):
+    code = marshal.loads(code)
+    globs = {'__builtins__': __builtins__}
```

Comment:
Why "globs"?
```suggestion
    builtins = {'__builtins__': __builtins__}
```

Reply:
It is a dictionary containing "global scope" of the function, the one available later on as `fun.__globals__` immutable attribute (you cannot replace the dict but can update its contents). If one called `globals()` inside the instantiated function this is the object that would be returned. So it is rather a coincidence that this dictionary contains only builtins at this point - all other global references (if any) are added at the set state stage to handle possible cyclic references.
I didn't call it globals on the other hand to avoid shadowing `globals()` inside `fun.__globals__`